BUILD.gn

```diff
@@ -0,0 +1,530 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/features.gni")
+import("//build/config/sanitizers/sanitizers.gni")
+import("//build/config/ui.gni")
+import("//build_overrides/v8.gni")
+import("//media/media_options.gni")
+import("//third_party/widevine/cdm/widevine.gni")
+import("//third_party/ffmpeg/ffmpeg_options.gni")
+import("//xwalk/build/version.gni")
+
+config("xwalk_runtime_config") {
+  # TODO(rakuco): Remove this once we stop supporting Ubuntu Precise or
+  # default to use_sysroot==1. This is only required because Precise's dbus
+  # 1.4.18 causes
+  # <https://bugs.chromium.org/p/chromium/issues/detail?id=263960#c6>.
+  if (is_clang && !use_sysroot) {
+    cflags = [ "-Wno-reserved-user-defined-literal" ]
+  }
+}
+
+source_set("xwalk_runtime") {
+  public_configs = [ "//build/config:feature_flags" ]
+  configs += [
+    # TODO(jschuh): crbug.com/167187 fix size_t to int truncations.
+    "//build/config/compiler:no_size_t_to_int_warning",
+    ":xwalk_runtime_config",
+  ]
+
+  defines = [
+    "XWALK_VERSION=\"$xwalk_version\"",
+    "CHROME_VERSION=\"$chrome_version\"",
+  ]
+
+  deps = [
+    ":generate_upstream_blink_version",
+    "//base:base",
+    "//base:base_static",
+    "//base:i18n",
+    "//base/third_party/dynamic_annotations",
+    "//cc",
+    "//components/autofill/content/browser",
+    "//components/autofill/content/renderer",
+    "//components/autofill/core/browser",
+    "//components/cdm/renderer",
+    "//components/devtools_http_handler",
+    "//components/resources:components_resources",
+    "//components/strings",
+    "//components/url_formatter",
+    "//components/user_prefs",
+    "//components/visitedlink/browser",
+    "//components/visitedlink/renderer",
+    "//content:content",
+    "//content/public/app:both",
+    "//content/public/browser",
+    "//content/public/child",
+    "//content/public/common",
+    "//content/public/utility",
+    "//gin:gin",
+    "//ipc:ipc",
+    "//media:media",
+    "//net:net",
+    "//net:net_resources",
+    "//skia:skia",
+    "//storage/browser:browser",
+    "//storage/common",
+    "//third_party/WebKit/public:blink",
+    "//third_party/boringssl:boringssl",
+    "//ui/base",
+    "//ui/gl",
+    "//ui/shell_dialogs",
+    "//ui/snapshot",
+    "//url:url",
+    "//v8",
+    "//xwalk/application:xwalk_application_lib",
+    "//xwalk/experimental:experimental",
+    "//xwalk/extensions:xwalk_extensions",
+    "//xwalk/resources:xwalk_pak",
+    "//xwalk/resources:xwalk_resources",
+    "//xwalk/resources:xwalk_resources_100_percent_pak",
+    "//xwalk/resources:xwalk_resources_200_percent_pak",
+    "//xwalk/resources:xwalk_resources_300_percent_pak",
+    "//xwalk/sysapps:sysapps",
+  ]
+  if (is_linux) {
```

Comment:
It makes more sense to put this at the bottom and group it with the other `is_linux` block you already have there.